osu.Game/Beatmaps/Drawables/Cards/Buttons/DownloadButton.cs

```diff
@@ -49,21 +56,44 @@ protected override void LoadComplete()
 
         private void updateState()
         {
-            this.FadeTo(state.Value != DownloadState.LocallyAvailable ? 1 : 0, BeatmapCard.TRANSITION_DURATION, Easing.OutQuint);
-
-            if (beatmapSet.Availability.DownloadDisabled)
+            switch (state.Value)
```

Comment:
Not sure about using a switch here. Open to converting to a series of `if-else` if that's preferred.

Reply:
I'm fine with the switch, but a bit less fine with pulling in the "undownloadable" case into the `NotDownloaded` label. Feels safer to me in the outside scope.

Note that this entire button does not show anyway if an undownloadable beatmap is locally available.

Reply:
I only put it in there because the test scene looks wrong at the moment otherwise (if somehow a download starts for the disabled beatmap, both icons can show at once).

Right now this is a side effect of the test cards sharing the same beatmap ID across multiple usages (one which is undownloadable), but not sure.

Reply:
This is what the test looks like with that check outside the `switch`:

https://user-images.githubusercontent.com/191335/143538514-594123a3-aecf-4b4f-9de8-0be07503bc10.mp4

Reply:
Hmm. I see. There's also the possibility of the import of an undownloadable map being initiated externally from outside the card, which will potentially run into the same scenario...

I'm now more convinced that doing this is now correct, if counterintuitive.